qiskit/primitives/containers/pub_result.py

```diff
@@ -0,0 +1,37 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Base Pub class
+"""
+
+from __future__ import annotations
+
+from qiskit.utils.optionals import HAS_PYDANTIC
+
+from .data_bin import DataBin
+from .dataclasses import frozen_dataclass
+
+if HAS_PYDANTIC:
+    from pydantic import Field
+else:
+    from dataclasses import field as Field
+
+
+@frozen_dataclass
+class PubResult:
```

Comment:
Is it not necessary to set `@HAS_PYDANTIC.require_in_instance`?

Reply:
If there is not pydantic, `dataclasses.dataclass` is used instead of `pydantic.dataclass`.